core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -699,6 +699,7 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
   private[tools] class MirrorMakerProducer(val producerProps: Properties) {
 
     val sync = producerProps.getProperty("producer.type", "async").equals("sync")
+    producerProps.remove("producer.type")
```

Comment:
Thanks for the PR. I suggest moving these two lines just before `MirrorMakerProducer` is created. We would then pass `sync` as a boolean parameter. The reason is that the code manipulating the properties would then be in the same place.